lib/pstore.rb

```diff
@@ -179,137 +393,112 @@ def fetch(name, default=PStore::Error)
+  #
   def path
     @filename
   end
 
+  # Exits the current transaction block after committing any changes
+  # specified in that block.
+  # See {Committing or Aborting}[rdoc-ref:PStore@Committing+or+Aborting].
   #
-  # Ends the current PStore#transaction, committing any changes to the data
-  # store immediately.
-  #
-  # == Example:
-  #
-  #  require "pstore"
-  #
-  #  store = PStore.new("data_file.pstore")
-  #  store.transaction do  # begin transaction
-  #    # load some data into the store...
-  #    store[:one] = 1
-  #    store[:two] = 2
-  #
-  #    store.commit        # end transaction here, committing changes
-  #
-  #    store[:three] = 3   # this change is never reached
-  #  end
-  #
-  # *WARNING*:  This method is only valid in a PStore#transaction.  It will
-  # raise PStore::Error if called at any other time.
-  #
+  # Raises an exception if called outside a transaction block.
   def commit
     in_transaction
     @abort = false
     throw :pstore_abort_transaction
   end
+
+  # Exits the current transaction block, ignoring any changes
+  # specified in that block.
+  # See {Committing or Aborting}[rdoc-ref:PStore@Committing+or+Aborting].
   #
-  # Ends the current PStore#transaction, discarding any changes to the data
-  # store.
-  #
-  # == Example:
-  #
-  #  require "pstore"
-  #
-  #  store = PStore.new("data_file.pstore")
-  #  store.transaction do  # begin transaction
-  #    store[:one] = 1     # this change is not applied, see below...
-  #    store[:two] = 2     # this change is not applied, see below...
-  #
-  #    store.abort         # end transaction here, discard all changes
-  #
-  #    store[:three] = 3   # this change is never reached
-  #  end
-  #
-  # *WARNING*:  This method is only valid in a PStore#transaction.  It will
-  # raise PStore::Error if called at any other time.
-  #
+  # Raises an exception if called outside a transaction block.
   def abort
     in_transaction
     @abort = true
     throw :pstore_abort_transaction
   end
 
+  # Defines a transaction block for the store.
+  # See {Transactions}[rdoc-ref:PStore@Transactions].
   #
-  # Opens a new transaction for the data store.  Code executed inside a block
-  # passed to this method may read and write data to and from the data store
-  # file.
-  #
-  # At the end of the block, changes are committed to the data store
-  # automatically.  You may exit the transaction early with a call to either
-  # PStore#commit or PStore#abort.  See those methods for details about how
-  # changes are handled.  Raising an uncaught Exception in the block is
-  # equivalent to calling PStore#abort.
-  #
-  # If _read_only_ is set to +true+, you will only be allowed to read from the
-  # data store during the transaction and any attempts to change the data will
-  # raise a PStore::Error.
+  # With argument +read_only+ as +false+, the block may contain any Ruby code,
+  # including calls to \PStore methods other #transaction.
```

Comment:
I think it's clear that the block may contain any Ruby code. `PStore#transaction` can't be nested regardless of of the value of `read_only`.

```suggestion
  # With argument +read_only+ as +false+, the block may both read and write to the \PStore.
```

Reply:
Fixed.